spark/src/main/java/org/apache/iceberg/spark/data/SparkParquetReaders.java

```diff
@@ -655,6 +653,11 @@ public int numElements() {
       return numElements;
     }
 
+    @Override
+    public int getNumElements() {
```

Comment:
Before, this was called `numElements`. Could you revert the name change?

Reply:
It looks like the function `numElements` is still there, just above this one

Reply:
It looks like the function `numElements` is still there (it's part of the catalyst abstract class), in the lines just above these ones.

But unless there's something going on with Spark, I agree with @rdblue.

I looked to see if `getNumElements` were needed on the Flink side, as Flink highly favors POJOs for performance reasons and then for style reasons they also tend to use POJO-y like names, but the Flink `ArrayData` interface uses `size` instead, so keeping the iceberg ReusableArrayData interface to use `numElements` seems like it would kill two birds with one stone by also fulfilling the catalyst contract.

Reply:
Yeah this is a good point, I'll simplify the interface so we don't have numElements and getNumElements.